spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteUpdate.scala

```diff
@@ -59,7 +62,7 @@ case class RewriteUpdate(spark: SparkSession) extends Rule[LogicalPlan] with Rew
       // so the first job uses DynamicFileFilter and the second one uses the underlying scan plan
       // both jobs share the same SparkMergeScan instance to ensure they operate on same files
       val matchingRowsPlanBuilder = scanRelation => Filter(cond, scanRelation)
-      val scanPlan = buildDynamicFilterScanPlan(spark, r.table, r.output, mergeBuilder, cond, matchingRowsPlanBuilder)
+      val scanPlan = buildDynamicFilterScanPlan(spark, r, r.output, mergeBuilder, cond, matchingRowsPlanBuilder)
```

Comment:
here I think we would be ok with using reflection to determine the buildDynamicFileterScanPlan to use, we would look to see which method signature is valid and use that. Since the return type is the same I think we are more less safe if we do that.

Reply:
Nvm I see this is our code so we could keep this change and alter buildDynamic to be compatible with both spark builds